lib/shared-method.js

```diff
@@ -269,35 +302,30 @@ function coerceAccepts(uarg, desc) {
     throw new badArgumentError(name + ' must be a number');
   }
 
-  // convert strings
-  if (actualType === 'string' && targetType !== 'any' && actualType !== targetType) {
-    // JSON.parse can throw, so catch this error.
-    try {
-      uarg = convertValueToTargetType(uarg, targetType);
-    } catch (e) {
-      var message = util.format('invalid value for argument \'%s\' of type ' +
-        '\'%s\': %s. Received type was %s. Error: %s',
-        name, targetType, uarg, typeof uarg, e.message);
-      throw new badArgumentError(message);
-    }
-  }
   return uarg;
 }
 
-function convertValueToTargetType(value, targetType) {
+function convertValueToTargetType(argName, value, targetType) {
   switch (targetType) {
     case 'string':
-      return value;
+      return String(value).valueOf();
     case 'date':
       return new Date(value);
     case 'number':
-      return Number(value);
+      return Number(value).valueOf();
     case 'boolean':
-      return Boolean(value);
+      return Boolean(value).valueOf();
```

Comment:
It returns a primitive type instead of a boxed (wrapped) object.

- `typeof Boolean(true) === 'object'`
- `typeof (Boolean(true).valueOf()) === 'boolean`